src/transformStateWithClones.js

```diff
@@ -7,7 +7,31 @@
  * @return {Object[]}
  */
 function transformStateWithClones(state, actions) {
-  // write code here
+  let currentState = { ...state };
+  const stateHistory = [];
+
+  for (const action of actions) {
+    switch (action.type) {
+      case 'clear':
+        currentState = {};
+        break;
+      case 'addProperties':
+        currentState = { ...currentState, ...action.extraData };
+        break;
+      case 'removeProperties':
+        currentState = { ...currentState };
+
+        for (const key of action.keysToRemove) {
+          delete currentState[key];
+        }
+        break;
+      default:
+        break;
+    }
+    stateHistory.push(currentState);
```

Comment:
You are pushing the 'currentState' object into 'stateHistory' after every action, but 'currentState' is a reference to an object that is being mutated on each iteration. This means that 'stateHistory' will contain multiple references to the same object. To fix this, you should push a copy of 'currentState' instead.